src/roots.jl

```diff
@@ -11,15 +11,39 @@ Base.size(x::Interval) = (1,)
 
 isinterior{N}(X::IntervalBox{N}, Y::IntervalBox{N}) = all(isinterior.(X, Y))
 
-struct RootSearchState{T <: Union{Interval,IntervalBox}}
-    working::Vector{T}
-    outputs::Vector{Root{T}}
+struct RootSearchState{V, VR}
+    working::V  # Should be a container of the form  CONT{T}
+    outputs::VR  # Should ba a container of root of the form CONT{Root{T}}
```

Comment:
I don't think it's necessary for `outputs` to be a fancy container -- isn't a standard list (vector) good enough, since there won't be complicated manipulations here?